lib/house.rb

```diff
@@ -0,0 +1,62 @@
+class House
+attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    rooms << room
+  end
+
+  def above_market_average?
+    if @price[1..-1].to_i > 500000
+      true
+    else
+      false
+    end
+  end
+
+  def rooms_from_category(category)
+    rooms_category = []
+    @rooms.each do |room|
+      if room.category == category
+        rooms_category << room
+      end
+    end
+    rooms_category
+  end
+
+  def area
+    all_room_areas = []
+    @rooms.each do |room|
+      all_room_areas << room.area
+    end
+    all_room_areas.sum
+  end
+
+  def details
+    details = {price: @price[1..-1].to_i, address: @address}
+  end
+
+  def price_per_square_foot
+    (@price[1..-1].to_i / area.to_f).round(2)
+  end
+
+  def rooms_sorted_by_area
+      @rooms.sort_by { |room| -room.area }
+  end
+
+  def rooms_by_category
+      @rooms.sort_by { |room| room.category }
```

Comment:
We actually need to build a hash here. This method is certainly the most difficult, but we could accomplish it with something like: 

```
rooms_by_categories = {}
@rooms.each do |room|
if rooms_by_categories[room.category]
rooms_by_categories[room.category] << room  
else  
rooms_by_categories[room.category] = [room]    
end
end
```